### ISSUE TITLE
H gate does not get decomposed correctly

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-terra#6411]
### Information

- **Qiskit Terra version**: latest (0.26)
- **Python version**:
- **Operating system**:

### What is the current behavior?
An H gate does not get decomposed correctly. It is easy to see this at optimization level 0:

![image](https://user-images.githubusercontent.com/1249193/118111500-c6c03280-b3b1-11eb-92c3-ec340d0b0d1c.png)

becomes

![image](https://user-images.githubusercontent.com/1249193/118111516-cf186d80-b3b1-11eb-9a78-d421f56d68ff.png)

when it should involve no more than one `sx` gate and two `rz` gates.

### What is the expected behavior?
An `H` gate gets decomposed into the correct minimal set of basis gates.

### Suggested solutions
It looks like the decomposition for `s` and `sdg` does not know they are simple z-axis rotations.

**Comment by Avhijit-codeboy** [comment]
Hi @nonhermitian !

This is an interesting issue. I would love to fix this, but this is my first time contributing to Qiskit, so a lot of guidance will be helpful.

**Comment by iamsantanubanerjee** [comment]
@nonhermitian Can you share the steps to reproduce this problem?

**Comment by nonhermitian** [comment]
Simply transpile the H gate using optimization level 0

**Comment by Avhijit-codeboy** [comment]
@nonhermitian I transpiled it with optimization level 0, but all I get is the `H` gate back
```
qc = QuantumCircuit(1)
qc.h(0)
a = transpile(qc,optimization_level = 0)
a.draw(output='mpl')
```
![image](https://user-images.githubusercontent.com/41890388/120893193-8e69e800-c62f-11eb-9935-cd5ad52c206c.png)

**Comment by nonhermitian** [comment]
Yeah sorry, you need to target an IBM backend or set the basis gates to be the same.

**Comment by Avhijit-codeboy** [comment]
By backend, do you mean the actual hardware or the simulator? `qasm_simulator` gave me the same output as above, and so did `basis_gates=['h']`

**Comment by nonhermitian** [comment]
A real backend

**Comment by Avhijit-codeboy** [comment]
I used `ibmq_belem` as the backend, and I got this error. What am I doing wrong? (the code is same as above)
```
Cannot unroll the circuit to the given basis, ['id', 'rz', 'sx', 'x', 'cx', 'reset']. No rule to expand instruction h.```

**Comment by nonhermitian** [comment]
This works on the latest release:

```python

qc = QuantumCircuit(1)
qc.h(0)

backend = provider.backend.ibmq_belem
trans_qc = transpile(qc, backend, optimization_level=0)
trans_qc.draw()
```

If it is not working on master than it is a bug.

**Comment by Avhijit-codeboy** [comment]
Ah..I see. My bad, I was checking on `PyPi` version

**Comment by nonhermitian** [comment]
The latest pypi version should work.

**Comment by Avhijit-codeboy** [comment]
The latest is version number `0.26.2`, and I checked my code on this version. But I wasn't able to reproduce the problem.

**Comment by nonhermitian** [comment]
The IBM Quantum website is at Qiskit `0.26.2` and give me:

![image](https://user-images.githubusercontent.com/1249193/120897369-c6533a00-c5f3-11eb-9b0f-6555b878f4b3.png)


I am not sure what you are seeing or not

**Comment by Avhijit-codeboy** [comment]
I see. Wonder why I'm not getting it. Btw, I tried it on a Jupyter Notebook.
This is the full code 
```
from qiskit import QuantumCircuit, Aer, IBMQ
from qiskit.compiler import transpile
from qiskit.providers.ibmq import least_busy

qc = QuantumCircuit(1)
qc.h(0)

provider = IBMQ.get_provider(hub='ibm-q')
small_devices = provider.backends(filters=lambda x: x.configuration().n_qubits == 5
                                   and not x.configuration().simulator)
backend = least_busy(small_devices)

a = transpile(qc,optimization_level = 0,backend = backend) # gives error
```

**Comment by Avhijit-codeboy** [comment]
@nonhermitian Can I take up this issue?

**Comment by nonhermitian** [comment]
Sure thing

**Comment by Avhijit-codeboy** [comment]
Thanks!

**Comment by kdk** [comment]
Thanks for your interest @Avhijit-codeboy ! Closing as this was already resolved as part of #4837 :

```
>>> qc = qk.QuantumCircuit(1)
>>> qc.h(0)
>>> qk.transpile(qc, basis_gates=['rz', 'sx'], optimization_level=0).draw()
global phase: π/4
     ┌─────────┐┌────┐┌─────────┐
q_0: ┤ RZ(π/2) ├┤ √X ├┤ RZ(π/2) ├
     └─────────┘└────┘└─────────┘
```

Please feel free to re-open if you're still seeing this behavior.
